File: two_pointers/11_container_with_most_water.py
def max_area(height: list[int]) -> int:
    l = 0
    r = len(height) - 1
    result = 0

    while l < r:
        width = r - l
        left_height = height[l]
        right_height = height[r]
        min_height = min(left_height, right_height)
        result = max(result, min_height * width)

        if left_height < right_height:
            l += 1
        else:
            r -= 1

    return result

File: two_pointers/11_container_with_most_water.go
func maxArea(height []int) int {
	res := 0
	l := 0
	r := len(height) - 1

	for l < r {
		width := r - l
		leftH := height[l]
		rightH := height[r]

		res = max(res, width*min(leftH, rightH))

		if leftH < rightH {
			l++
		} else {
			r--
		}

	}
	return res
}
